athena-opensearch/src/main/java/com/amazonaws/athena/connectors/opensearch/OpensearchMetadataHandler.java

```diff
@@ -0,0 +1,426 @@
+            throws Exception
+    {
+        LOGGER.debug("doGetTable: enter - " + getTableRequest);
+
+        Schema schema = null;
+        Schema partitionSchema = getPartitionSchema(getTableRequest.getCatalogName());
+
+        // Look at GLUE catalog first.
+        try {
+            if (!this.awsGlue) {
+                schema = glueHandler.doGetTable(blockAllocator, getTableRequest).getSchema();
+                LOGGER.info("glueDoGetTable: Retrieved schema for table[{}] from AWS Glue.", getTableRequest.getTableName());
+            }
+        }
+        catch (Exception error) {
+            LOGGER.warn("glueDoGetTable: Unable to retrieve table[{}:{}] from AWS Glue.",
+            getTableRequest.getTableName().getSchemaName(),
+            getTableRequest.getTableName().getTableName(),
+                    error);
+        }
+
+        if (schema == null) {
+            try (Connection connection = jdbcConnectionFactory.getConnection(getCredentialProvider())) {
+                schema = getSchema(connection, getTableRequest.getTableName(), partitionSchema);
+            }
+        }
+
+        return new GetTableResponse(getTableRequest.getCatalogName(), getTableRequest.getTableName(),
+                (schema == null) ? SchemaBuilder.newBuilder().build() : schema, partitionSchema.getFields().stream().map(Field::getName).collect(Collectors.toSet()));
+    }
+
+    protected Schema getSchema(Connection jdbcConnection, TableName tableName, Schema partitionSchema)
+            throws SQLException
+    {
+        LOGGER.info("Getting Schema for table: {} ", tableName);
+        SchemaBuilder schemaBuilder = SchemaBuilder.newBuilder();
+        try (ResultSet resultSet = getColumns(jdbcConnection.getCatalog(), tableName, jdbcConnection.getMetaData())) {
+            boolean found = false;
+            while (resultSet.next()) {
+                int dataType = resultSet.getInt("DATA_TYPE");
+                int columnSize = resultSet.getInt("COLUMN_SIZE");
+                int decimalDigits = resultSet.getInt("DECIMAL_DIGITS");
+                ArrowType columnType = JdbcArrowTypeConverter.toArrowType(
+                        dataType,
+                        columnSize,
+                        decimalDigits,
+                        configOptions);
+                String columnName = resultSet.getString("COLUMN_NAME");
+                if (columnType != null && SupportedTypes.isSupported(columnType)) {
+                    if (columnType instanceof ArrowType.List) {
+                        schemaBuilder.addListField(columnName, getArrayArrowTypeFromTypeName(
+                                resultSet.getString("TYPE_NAME"),
+                                resultSet.getInt("COLUMN_SIZE"),
+                                resultSet.getInt("DECIMAL_DIGITS")));
+                    }
+                    else if (columnType instanceof ArrowType.Struct) {
+                        continue;
```

Comment:
i remember you said that structs are flattened into single fields containing ".", but then what circumstance will return a column of type Struct where you need this continue case?